components/ShadowElevations/examples/ShadowElevationsTypicalUseExample.m

```diff
@@ -124,10 +201,65 @@ - (void)viewDidLoad {
-      UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth;
   [self.view addSubview:_shadowsView];
+  
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+  if (@available(iOS 11.0, *)) {
+    self.shadowsView.autoresizingMask =
+    UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+  } else {
+#endif
+    _shadowsView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self setupShadowsViewConstraints];
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+  }
+#endif
+}
+
+- (void)setupShadowsViewConstraints {
+  [NSLayoutConstraint constraintWithItem:self.shadowsView
+                               attribute:NSLayoutAttributeTop
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self.topLayoutGuide
+                               attribute:NSLayoutAttributeBottom
+                              multiplier:1.0
+                                constant:0].active = YES;
+  
+  [NSLayoutConstraint constraintWithItem:self.shadowsView
+                               attribute:NSLayoutAttributeLeading
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self.view
+                               attribute:NSLayoutAttributeLeading
+                              multiplier:1.0
+                                constant:0].active = YES;
+  
+  [NSLayoutConstraint constraintWithItem:self.shadowsView
+                               attribute:NSLayoutAttributeWidth
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self.view
+                               attribute:NSLayoutAttributeWidth
+                              multiplier:1.0
+                                constant:0].active = YES;
+  
+  [NSLayoutConstraint constraintWithItem:self.shadowsView
+                               attribute:NSLayoutAttributeBottom
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self.bottomLayoutGuide
+                               attribute:NSLayoutAttributeTop
+                              multiplier:1.0
+                                constant:0].active = YES;
+}
+
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+- (void)viewSafeAreaInsetsDidChange {
+  [super viewSafeAreaInsetsDidChange];
+  CGRect insetedShadowViewFrame = CGRectMake(self.view.bounds.origin.x,
+                                             self.view.bounds.origin.y + self.view.safeAreaInsets.top,
+                                             self.view.bounds.size.width - self.view.safeAreaInsets.left - self.view.safeAreaInsets.right,
+                                             self.view.bounds.size.height - self.view.safeAreaInsets.top - self.view.safeAreaInsets.bottom);
+  self.shadowsView.frame = insetedShadowViewFrame;
 }
+#endif
```

Comment:
Not sure if there is a better method I can utilize to layout with safeArea in the view update life cycle here.